src/e2e/utils/helpers.ts

```diff
@@ -45,6 +45,29 @@ export const getBaseUrl = () => {
   return ENV_URLS[process.env.E2E_TEST_ENV as ENV] || ENV_URLS.local;
 };
 
+export const delay = async (time: number) => {
+  return new Promise(function (resolve) {
+    setTimeout(resolve, time);
+  });
+};
+
+export const waitForUrlOrTimeout = async (
```

Comment:
Isn't this basically [`waitForURL`](https://playwright.dev/docs/api/class-page#page-wait-for-url)? And if not, could you add a comment why it's needed and why `waitForURL` doesn't suffice?

Reply:
this is an issue with the click of free scan, that sometimes does not redirect to the scan page, will add a comment here

Reply:
I might be confused re: `timeout` vs `tries`+`delay`. Looks like the usage above is:
```js
const freeScanRedirect = await waitForUrlOrTimeout(page, "/scan", 5000);
```

But the max timeout value will maybe, kinda, always be 5000ms regardless. So an explicit `timeout` of 10_000ms would still timeout after 5s because we have a max 10 tries w/ a 500ms delay between checks. If we exceed the specified `timeout` OR the number of attempts is 10+, the `while` loop would exit and the function would return `void`|`undefined`|whatever.

Anyways, not a blocker, and possibly not even interesting/relevant, but just thought I'd throw it out there since it might be a gotcha if we end up using this function more, or somebody tries changing the /scan page timeout in the other file and wonders why it still exits at 5s despite a timeout of 40s... 🤷

Reply:
noted, will update this in the following ticket to use the 5000/or whatever is provided